boa_engine/src/builtins/intl/mod.rs

```diff
@@ -144,3 +147,512 @@ impl Intl {
+            if let Some(entry) = maybe_entry {
+                // 2. Let requestedValue be entry.[[Value]].
+                let requested_value = &entry.value;
+
+                // 3. If requestedValue is not the empty String, then
+                if !requested_value.is_empty() {
+                    // a. If keyLocaleData contains requestedValue, then
+                    if key_locale_data.contains(requested_value) {
+                        // i. Let value be requestedValue.
+                        value = JsValue::String(JsString::new(requested_value));
+                        // ii. Let supportedExtensionAddition be the string-concatenation
+                        // of "-", key, "-", and value.
+                        supported_extension_addition =
+                            JsString::concat_array(&["-", key, "-", requested_value]);
+                    }
+                // 4. Else if keyLocaleData contains "true", then
+                } else if key_locale_data.contains(&JsString::new("true")) {
+                    // a. Let value be "true".
+                    value = JsValue::String(JsString::new("true"));
+                    // b. Let supportedExtensionAddition be the string-concatenation of "-" and key.
+                    supported_extension_addition = JsString::concat_array(&["-", key]);
+                }
+            }
+        }
+
+        // i. If options has a field [[<key>]], then
+        if options.properties.contains_key(key) {
+            // i. Let optionsValue be options.[[<key>]].
+            // TODO ii. Assert: Type(optionsValue) is either String, Undefined, or Null.
+            let mut options_value = options
+                .properties
+                .get(key)
+                .unwrap_or(&JsValue::undefined())
+                .clone();
+
+            // iii. If Type(optionsValue) is String, then
+            if options_value.is_string() {
+                // TODO 1. Let optionsValue be the string optionsValue after performing the
+                // algorithm steps to transform Unicode extension values to canonical syntax
+                // per Unicode Technical Standard #35 LDML § 3.2.1 Canonical Unicode Locale
+                // Identifiers, treating key as ukey and optionsValue as uvalue productions.
+
+                // TODO 2. Let optionsValue be the string optionsValue after performing the
+                // algorithm steps to replace Unicode extension values with their canonical
+                // form per Unicode Technical Standard #35 LDML § 3.2.1 Canonical Unicode
+                // Locale Identifiers, treating key as ukey and optionsValue as uvalue
+                // productions.
+
+                // 3. If optionsValue is the empty String, then
+                let options_val_str = options_value
+                    .to_string(context)
+                    .unwrap_or_else(|_| JsString::new(""))
+                    .to_string();
```

Comment:
I think you should be able to do `if let Some(options_val_str) = options_value.as_string()` here.

Reply:
[Applied](https://github.com/boa-dev/boa/pull/2036/files#diff-1dd4af22d3a6142a8060d056e9d0489d84d8a4d9c68a8f6e7b359b8d3ab35499R613).